test/gui/shared/scripts/pageObjects/AccountConnectionWizard.py

```diff
@@ -25,34 +25,57 @@ class AccountConnectionWizard:
         "visible": 1,
         "window": names.error_QMessageBox,
     }
+    URL_LOCKED = {
+        "name": "urlLabel",
+        "type": "QLabel",
+        "visible": 1,
+        "window": names.owncloudWizard_OCC_OwncloudWizard,
+    }
+    ERROR_LABEL = {
+        "name": "errorLabel",
+        "type": "QLabel",
+        "visible": 1,
+        "window": names.owncloudWizard_OCC_OwncloudWizard,
+    }
 
     def __init__(self):
         pass
 
     def sanitizeFolderPath(self, folderPath):
         return folderPath.rstrip("/")
 
-    def addAccount(self, context):
-        server, user, password, localfolder = getClientDetails(context)
-
+    def addServer(self, serverAddress):
         squish.mouseClick(squish.waitForObject(self.SERVER_ADDRESS_BOX))
-        squish.type(squish.waitForObject(self.SERVER_ADDRESS_BOX), server)
+        squish.type(squish.waitForObject(self.SERVER_ADDRESS_BOX), serverAddress)
         squish.clickButton(squish.waitForObject(self.NEXT_BUTTON))
+
+    def addAccount(self, context):
+        clientDetails = getClientDetails(context)
+        self.addServer(clientDetails['server'])
         squish.mouseClick(squish.waitForObject(self.SERVER_ADDRESS_BOX))
-        squish.type(squish.waitForObject(self.USERNAME_BOX), user)
+        squish.type(squish.waitForObject(self.USERNAME_BOX), clientDetails['user'])
         squish.type(squish.waitForObject(self.USERNAME_BOX), "<Tab>")
-        squish.type(squish.waitForObject(self.PASSWORD_BOX), password)
+        squish.type(squish.waitForObject(self.PASSWORD_BOX), clientDetails['password'])
         squish.clickButton(squish.waitForObject(self.NEXT_BUTTON))
+
+    def selectSyncFolder(self, context):
+        clientDetails = getClientDetails(context)
+
         try:
             squish.clickButton(squish.waitForObject(self.ERROR_OK_BUTTON))
         except LookupError:
             pass
         squish.clickButton(squish.waitForObject(self.SELECT_LOCAL_FOLDER))
         squish.mouseClick(squish.waitForObject(self.DIRECTORY_NAME_BOX))
-        squish.type(squish.waitForObject(self.DIRECTORY_NAME_BOX), localfolder)
+        squish.type(
+            squish.waitForObject(self.DIRECTORY_NAME_BOX), clientDetails['localfolder']
+        )
         squish.clickButton(squish.waitForObject(self.CHOOSE_BUTTON))
         test.compare(
             str(squish.waitForObjectExists(self.SELECT_LOCAL_FOLDER).text),
-            self.sanitizeFolderPath(localfolder),
+            self.sanitizeFolderPath(clientDetails['localfolder']),
         )
         squish.clickButton(squish.waitForObject(self.FINISH_BUTTON))
+
+    def isUrlLocked(self):
```

Comment:
this function seems to check the presence of an element. So better to rename the function to something else.
`isUrlLocked` is kind of misleading